src/common/common-types.ts

```diff
@@ -308,12 +308,48 @@ export interface Feature {
     readonly description: string;
 }
 
+export interface ISetting {
```

Comment:
No `IName`. Common convention is to use `NameBase` and make the interface non-exported.
```suggestion
interface SettingBase {
```